### PR TITLE
Feature/polytopes/catalan_solid

### PR DESCRIPTION
**Pull request by OguzhanYueruek**
Added Catalan solids construction, added tests for archimedean and catalan solids.

### CODE REVIEW

**Comment by benlorenz** [comment]
Adding `CompilerSupportLibraries_jll` did fix that `GLIBCXX` error for julia 1.6 and 1.7 (on ubuntu) but after removing it again the error is still gone and the wrapper compilation does succeed: https://github.com/oscar-system/Oscar.jl/runs/3292407484#step:5:119

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 1]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
 """
 archimedean_solid(s::String) = Polyhedron(Polymake.polytope.archimedean_solid(s))
 
+
+@doc Markdown.doc"""
+
+    catalan_solid(s)
+
+Construct a Catalan solid with the name given by String `s` from the list
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl]:
```suggestion
    catalan_solid(s::String)

Construct a Catalan solid with the name `s` from the list
```

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 2]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
...
+below.  The polytopes are realized with floating point numbers and thus not
+exact; Vertex-facet-incidences are correct in all cases.
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl]:
```suggestion
below.  The polytopes are realized with floating point coordinates and thus are not
exact. However, vertex-facet-incidences are correct in all cases.
```

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 3]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
...
+
+# Arguments
+- `s::String`: The name of the desired Catalan solid.
+
+    Possible values:
+
+  String s the name of the desired Catalan solid
+    Possible values:
+
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl]:
```suggestion
The following strings are admissible values for `s`:
```

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 4]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
...
+      "deltoidal_icositetrahedron" : Deltoidal Icositetrahedron.
+          Dual polytope to the rhombicubaoctahedron, made of 24 kite facets.
+      "pentakis_dodecahedron" : Pentakis dodecahedron.
+          Dual polytope to the truncated icosahedron, made of 60 isosceles
+          triangular facets.
+      "deltoidal_hexecontahedron" : Deltoidal hexecontahedron.
+          Dual polytope to the rhombicosidodecahedron, made of 60 kite facets.
+      "disdyakis_triacontahedron" : Disdyakis triacontahedron.
+          Dual polytope to the truncated icosidodecahedron, made of 120 scalene
+          triangular facets.
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl:606]: I wonder how this is rendered in practice? Perhaps turning it into a markdown list would be nicer? But I didn't check the output, so perhaps your way *is* better?

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 5]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
...
+
+
+# Examples
+```julia-repl
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl]: Please turn this into a `jldoctest` instead if at all possible.

File: test/Polytopes/runtests.jl [hunk 6]
```diff
@@ -1,6 +1,11 @@
 const pm = Polymake
 
 using Oscar
+
+if (haskey(ENV, "GITHUB_ACTIONS"))
+    Polymake.shell_execute(raw"$Verbose::cpp=3;")
```
Comment (fingolfin) [inline test/Polytopes/runtests.jl:10]: What is this about?

File: src/Polytopes/Polyhedron/standard_constructions.jl [hunk 7]
```diff
@@ -524,6 +524,74 @@ julia> nfacets(T)
...
+julia> T = catalan_solid("triakis_tetrahedron")
+A polyhedron in ambient dimension 3
+
+julia> sum([nvertices(F) for F in faces(T, 2)] .== 3)
```
Comment (fingolfin) [inline src/Polytopes/Polyhedron/standard_constructions.jl]: This seems to be counting the vertices that are triangles? if so, I'd find this clearer:
```suggestion
julia> count(F -> nvertices(F) == 3, faces(T, 2))
```

File: test/Polytopes/Polyhedron.jl [hunk 8]
```diff
@@ -52,7 +52,15 @@
 
     @testset "standard_constructions" begin
         p = upper_bound_theorem(4,8)
+        A = archimedean_solid("cuboctahedron")
         @test p.pm_polytope.F_VECTOR == [8, 28, 40, 20]
+        @test sum([nvertices(F) for F in faces(A, 2)] .== 3) == 8
```
Comment (fingolfin) [inline test/Polytopes/Polyhedron.jl]:
```suggestion
        @test p.pm_polytope.F_VECTOR == [8, 28, 40, 20]
        A = archimedean_solid("cuboctahedron")
        @test count(F -> nvertices(F) == 3, faces(A, 2)) == 8
```

File: test/Polytopes/Polyhedron.jl [hunk 9]
```diff
@@ -52,7 +52,15 @@
...
+        # due to GLIBCXX issues with the outdated julia-shipped libstdc++
+        # we run this only where recent CompilerSupportLibraries are available
+        if VERSION >= v"1.6"
+            C = catalan_solid("triakis_tetrahedron")
+            @test sum([nvertices(F) for F in faces(C, 2)] .== 3) == 12
```
Comment (fingolfin) [inline test/Polytopes/Polyhedron.jl]:
```suggestion
            @test count(F -> nvertices(F) == 3, faces(C, 2)) == 12
```

File: src/Polytopes/Polyhedron/standard_constructions.jl (hunk 4 above)
Comment (fingolfin) [reply src/Polytopes/Polyhedron/standard_constructions.jl:606]: Ah, I just realized that `archimedean_solid` is documented in a similar way; and in fact, the "list" is a big code block. Not very nice, I'd definitely prefer a proper Markdown list over that.

File: test/Polytopes/runtests.jl (hunk 6 above)
Comment (benlorenz) [reply test/Polytopes/runtests.jl:10]: This is a debug flag from polymake to add some debugging output when polymake wrappers are compiled during the tests. It was helpful to identify one more GLIBCXX issue. I think it should stay in there to make it easier to analyze other issues with polymake compilation at runtime (we also have this in the Polymake.jl tests).

**Comment by benlorenz** [comment]
> Adding `CompilerSupportLibraries_jll` did fix that `GLIBCXX` error for julia 1.6 and 1.7 (on ubuntu) but after removing it again the error is still gone and the wrapper compilation does succeed: https://github.com/oscar-system/Oscar.jl/runs/3292407484#step:5:119

On my test-branch where I removed the CompilerSupportLibraries_jll again I did get another error after a few re-runs (https://github.com/oscar-system/Oscar.jl/runs/3293977146). Since CompilerSupportLibraries_jll is an (indirect) dependency anyway, keeping it in Oscar directly just makes sure it is loaded earlier and seems to fix this `GLIBCXX_3.4.26` issue at least for 1.6 and 1.7 (probably related to some improvements/updates via stdlib jll packages). I will also do some more re-runs and more local tests (but so far I couldn't reproduce this locally with julia 1.6).

File: test/Polytopes/runtests.jl (hunk 6 above)
Comment (fingolfin) [reply test/Polytopes/runtests.jl:10]: That's fine by me, but then a comment should be added to the code code explaining this

File: src/Polytopes/Polyhedron/standard_constructions.jl (hunk 4 above)
Comment (OguzhanYueruek) [reply src/Polytopes/Polyhedron/standard_constructions.jl:606]: I was also thinking about changing how does the list look like for achimedean_solid, but then convinced that it looked okay. Now that you brought the idea up, I think it makes sense to turn this into a Markdown.

File: src/Polytopes/Polyhedron/standard_constructions.jl (hunk 5 above)
Comment (OguzhanYueruek) [reply src/Polytopes/Polyhedron/standard_constructions.jl]: I have taken care of this for the Catalan solids in this pull request. The rest are taken care of in the pull request #611.

File: src/Polytopes/Polyhedron/standard_constructions.jl (hunk 4 above)
Comment (OguzhanYueruek) [reply src/Polytopes/Polyhedron/standard_constructions.jl:606]: This is taken care of for Catalan solids in this pull request. The same problem for Archimedean solids have been resolved in the pull request #611.

Comment (fingolfin) [reply src/Polytopes/Polyhedron/standard_constructions.jl:606]: Thanks @OguzhanYueruek

**Review by fingolfin — APPROVED** [review]
This looks fine to me, but it is still marked as draft, so I'll leave it to @lkastner @benlorenz and who else to decide whether and when to merge this :-)

**Comment by fingolfin** [comment]
It seems the doctests may be failing.

Also, perhaps this could be squashed down into 1-2 commits (one for @benlorenz change, one for @OguzhanYueruek's work). 18 commits for this PR seems a bit overkill, and makes it harder to `git bisect` the changes.

**Comment by OguzhanYueruek** [comment]
> It seems the doctests may be failing.
> 
> Also, perhaps this could be squashed down into 1-2 commits (one for @benlorenz change, one for @OguzhanYueruek's work). 18 commits for this PR seems a bit overkill, and makes it harder to `git bisect` the changes.

The issue regading doctest should be fixed now. I mark this as ready for now. There is still a problem regarding tests that @benlorenz is aware of, he can decide if/when to accept the merge.

**Comment by benlorenz** [comment]
I can do the squash later, and I hope I have a fix ready later today (I need to run the tests for a while to make sure and will need to build new polymake_jll binaries).

**Comment by benlorenz** [comment]
My fix for the wrappers did fix it for ubuntu-latest but unfortunately seems to have broken it for macos ...

**Comment by benlorenz** [comment]
To make the wrapper compilation work properly I added a small patch to `polymake_jll` which appeared as version `400.400.1+2`.

**Pull request merged**